frontend/src/metabase/visualizations/visualizations/Text/Text.jsx

```diff
@@ -94,10 +99,46 @@ export default class Text extends Component {
   preventDragging = e => e.stopPropagation();
 
   render() {
-    const { className, gridSize, settings, isEditing, isPreviewing } =
-      this.props;
+    const {
+      className,
+      dashboard,
+      dashcard,
+      gridSize,
+      settings,
+      isEditing,
+      isPreviewing,
+      parameterValues,
+    } = this.props;
     const isSingleRow = gridSize && gridSize.height === 1;
 
+    let parametersByTag = {};
+    if (dashcard && dashcard.parameter_mappings) {
+      parametersByTag = dashcard.parameter_mappings.reduce((acc, mapping) => {
+        const tagId = mapping.target[1];
+        const parameter = dashboard.parameters.find(
+          p => p.id === mapping.parameter_id,
+        );
+        if (parameter) {
+          const parameterValue = parameterValues[parameter.id];
+          return {
+            ...acc,
+            [tagId]: { ...parameter, value: parameterValue },
+          };
+        } else {
+          return acc;
+        }
+      }, {});
+    }
+
+    let content = settings["text"];
+    if (!_.isEmpty(parametersByTag)) {
```

Comment:
```suggestion
    if (! _.isEmpty(parametersByTag)) {
```
nit: I believe `!` here is the `not` operation? if so having a space between will makes things less cryptic IMO.

Reply:
I see a lot of existing examples in the code of `!_.isEmpty` but none of `! _.isEmpty` so I presume the former is the preferred style